draft-ietf-mls-protocol.md

```diff
@@ -1518,15 +1518,24 @@ to other members in the group via an UpdatePath message (see {{update-paths}}) .
 All other group members then apply the key material in the UpdatePath to their
 own local tree state to derive the group's now-updated shared secret.
 
-To begin, the generator of the UpdatePath updates its leaf
-KeyPackage and its direct path to the root with new secret values.  The
-HPKE leaf public key within the KeyPackage MUST be derived from a freshly
-generated HPKE secret key to provide post-compromise security.
-
-The generator of the UpdatePath starts by sampling a fresh random value called
-"leaf_secret", and uses the leaf_secret to generate their leaf HPKE key pair
-(see {{key-packages}}) and to seed a sequence of "path secrets", one for each
-ancestor of its leaf. In this setting,
+To begin with, the generator of the UpdatePath updates its leaf's KeyPackage and
```

Comment:
I'm aware that this is not the style this section was written in previously, but when trying to implement an algorithm like this, I found it very helpful in the past if the description is a bit more structured. An example of this is the parent hash verification algorithm that (in the shape of bullet points) has pretty clear instructions on what should happen when.

Reply:
I agree. But I also think that this text could use a general improvement. I created a separate PR for this #593